src/iop/channelmixerrgb.c

```diff
@@ -747,7 +749,10 @@ static inline void loop_switch(const float *const restrict in, float *const rest
     /* FROM HERE WE ARE MANDATORILY IN XYZ - DATA IS IN temp_one */
 
     // Gamut mapping happens in XYZ space no matter what
-    gamut_mapping(temp_one, gamut, clip, temp_two);
+    if(do_gamut_mapping)
+      gamut_mapping(temp_one, gamut, clip, temp_two);
+    else
+      for(size_t c = 0; c < 3; ++c) temp_two[c] = temp_one[c];
```

Comment:
Why not use `for_each_channel`, or even more succinctly, `copy_pixel(temp_two,temp_one)`?  Copying three elements won't vectorize.

Reply:
because we are already in a SIMD loop, and one of the compilers will fail on nesting SIMD loops inside SIMD loops.

Reply:
Gotcha.  In that case, I would recommend using DT_PIXEL_SIMD_CHANNELS instead of 3, which is what `for_each_channel` uses internally.

Reply:
Done